lib/api_resources/servers.js

```diff
@@ -525,7 +526,26 @@ ServerResource.prototype.deviceAction = function(env, next) {
 
     var cb = function(err) {
       if (err) {
-        env.response.statusCode = 500;
+        var properties = {};
+        var statusCode = 500;
+
+        if(err instanceof ActionError) {
+          statusCode = err.statusCode;
+          properties = err.properties;
+        } else if (err instanceof Error) {
+          properties.message = err.message;
+        } else if(typeof error === 'string') {
+          properties.message = error
+        }
+
+        env.response.statusCode = statusCode;
+        env.response.body = {
+          class: ['action-error'],
+          properties: err.properties,
```

Comment:
This should just be `properties` not `err.properties` Message isn't getting passed in the two other cases.